pkg/crc/api/handlers.go

```diff
@@ -0,0 +1,102 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/crc/pkg/crc/validation"
+	"github.com/code-ready/crc/pkg/crc/version"
+)
+
+func statusHandler(_ ArgsType) string {
+	statusConfig := machine.ClusterStatusConfig{Name: constants.DefaultName}
+	clusterStatus, err := machine.Status(statusConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	return encodeStructToJson(clusterStatus)
+}
+
+func stopHandler(_ ArgsType) string {
+	stopConfig := machine.StopConfig{
+		Name:  constants.DefaultName,
+		Debug: true,
+	}
+	commandResult, err := machine.Stop(stopConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	return encodeStructToJson(commandResult)
+}
+
+func startHandler(_ ArgsType) string {
+	startConfig := machine.StartConfig{
+		Name:          constants.DefaultName,
+		BundlePath:    crcConfig.GetString(config.Bundle.Name),
+		VMDriver:      crcConfig.GetString(config.VMDriver.Name),
+		Memory:        crcConfig.GetInt(config.Memory.Name),
+		CPUs:          crcConfig.GetInt(config.CPUs.Name),
+		NameServer:    crcConfig.GetString(config.NameServer.Name),
+		GetPullSecret: getPullSecretFileContent,
+		Debug:         true,
+	}
+	status, err := machine.Start(startConfig)
+	if err != nil {
+		logging.Error(err.Error())
+		return "Failed"
+	}
+	return encodeStructToJson(status)
+}
+
+func versionHandler(_ ArgsType) string {
+	v := &machine.VersionResult{
+		CrcVersion:       fmt.Sprintf("%s+%s", version.GetCRCVersion(), version.GetCommitSha()),
+		OpenshiftVersion: version.GetBundleVersion(),
+		Success:          true,
+	}
+	return encodeStructToJson(v)
+}
+
+func getPullSecretFileContent() (string, error) {
+	data, err := ioutil.ReadFile(crcConfig.GetString(config.PullSecretFile.Name))
+	if err != nil {
+		return "", err
+	}
+	pullsecret := string(data)
+	if err := validation.ImagePullSecret(pullsecret); err != nil {
+		return "", err
+	}
+	return pullsecret, nil
+}
+
+func deleteHandler(_ ArgsType) string {
+	delConfig := machine.DeleteConfig{Name: constants.DefaultName}
+	r, err := machine.Delete(delConfig)
+	if err != nil {
+		return "Failed"
+	}
+	return encodeStructToJson(r)
+}
+
+func webconsoleURLHandler(_ ArgsType) string {
+	consoleConfig := machine.ConsoleConfig{Name: constants.DefaultName}
+	r, err := machine.GetConsoleURL(consoleConfig)
+	if err != nil {
+		return "Failed"
+	}
+	return encodeStructToJson(r)
+}
+
+func encodeStructToJson(v interface{}) string {
+	s, err := json.Marshal(v)
+	if err != nil {
+		return "Failed"
```

Comment:
@cfergeau something like `{"error": "Failed to encode information to json"}`?

I was not creating JSON requests for failed scenarios because on the tray side we assume a structure of the expected response, e.g when the start action is performed, the tray tries to unmarshal the returned string into a `StartResult` structure by first checking if the returned string is not `Failed`.

Reply:
In my opinion, using a `CommandError` struct for that which would marshall to what you suggest `{"error": "Failed to encode information to json"}` would feel slightly more robust/future proof than a hardcoded `Failed` string match.

Reply:
We can definitely do that, but a simple string keeps things simpler also in the tray side. Will you consider it as something that can be done as an improvement in a follow up patch?

Reply:
Yes this can be done as an improvement, but the less protocol changes we have to make, the better imo.

Reply:
Thanks, i do agree but we need to gather early feedback.